lib/ain-ocean/src/api/rawtx.rs

```diff
@@ -1,27 +1,148 @@
+use std::{str::FromStr, sync::Arc};
 
+use ain_macros::ocean_endpoint;
 use axum::{
+    extract::{Json, Path},
     routing::{get, post},
-    Router,
+    Extension, Router,
 };
-use defichain_rpc::{Client, RpcApi};
-use super::path::Path;
+use bitcoin::Txid;
+use defichain_rpc::{json::Bip125Replaceable, RpcApi};
+use rust_decimal::{prelude::ToPrimitive, Decimal};
 
-async fn send_rawtx() -> String {
-    "Sending raw transaction".to_string()
+use super::{response::Response, AppContext};
+use crate::{
+    error::ApiError,
+    model::{
+        default_max_fee_rate, MempoolAcceptResult, RawTransaction, RawTxDto, TransctionDetails,
+        WalletTxInfo,
+    },
+    Error, Result,
+};
+
+#[ocean_endpoint]
+async fn send_rawtx(
+    Extension(ctx): Extension<Arc<AppContext>>,
+    Json(raw_tx_dto): Json<RawTxDto>,
+) -> Result<String> {
+    println!("{:?}", raw_tx_dto.hex.clone());
+    validate(raw_tx_dto.hex.clone())?;
+    let mut max_fee = Some(default_max_fee_rate().unwrap().to_sat());
+    if let Some(fee_rate) = raw_tx_dto.max_fee_rate {
+        let sat_per_bitcoin = Decimal::new(100_000_000, 0);
+        let fee_in_satoshis = fee_rate * sat_per_bitcoin;
+        max_fee = fee_in_satoshis.round().to_u64();
+    }
+    let trx = defichain_rpc::RawTx::raw_hex(raw_tx_dto.hex);
+    match ctx.client.send_raw_transaction(trx, max_fee).await {
+        Ok(tx_hash) => Ok(tx_hash.to_string()),
+        Err(e) => {
+            eprintln!("Failed to send raw transaction: {:?}", e);
+            Err(Error::RpcError(e))
+        }
+    }
+}
+#[ocean_endpoint]
+async fn test_rawtx(
+    Extension(ctx): Extension<Arc<AppContext>>,
+    Json(raw_tx_dto): Json<RawTxDto>,
+) -> Result<Response<Vec<MempoolAcceptResult>>> {
+    let trx = defichain_rpc::RawTx::raw_hex(raw_tx_dto.hex);
+    let mut max_fee = Some(default_max_fee_rate().unwrap().to_sat());
+    if let Some(fee_rate) = raw_tx_dto.max_fee_rate {
+        let sat_per_bitcoin = Decimal::new(100_000_000, 0);
+        let fee_in_satoshis = fee_rate * sat_per_bitcoin;
+        max_fee = fee_in_satoshis.round().to_u64();
+    }
+    match ctx.client.test_mempool_accept(&[trx], max_fee).await {
+        Ok(mempool_tx) => {
+            let results = mempool_tx
+                .into_iter()
+                .map(|tx_result| MempoolAcceptResult {
+                    txid: tx_result.txid,
+                    allowed: tx_result.allowed,
+                    reject_reason: tx_result.reject_reason,
+                    vsize: tx_result.vsize,
+                    fees: tx_result.fees.map(|f| f.base),
+                })
+                .collect::<Vec<MempoolAcceptResult>>();
+            Ok(Response::new(results))
+        }
+        Err(e) => {
+            eprintln!("Failed to send raw transaction: {:?}", e);
+            Err(Error::RpcError(e))
+        }
+    }
 }
 
-async fn test_rawtx() -> String {
-    "Testing raw transaction".to_string()
+#[ocean_endpoint]
+async fn get_raw_tx(
```

Comment:
test failed

Reply:
This is bcoz of `jf` `model RawTrasnction` is different compare what is return from `rust_defi_rpc`

Reply:
https://github.com/BirthdayResearch/jellyfishsdk/blob/cb1c46722c9495dcfffd1e9f8c6d9a684d4dc7c9/apps/whale-api/src/module.api/rawtx.controller.ts#L99